frontend/integration-tests/views/dashboard.view.ts

```diff
@@ -11,3 +11,6 @@ export const activityCard = $('[data-test-id="activity-card"]');
 export const eventsPauseButton = $('[data-test-id="events-pause-button"]');
 export const launcherCard = $('[data-test-id="launcher-card"]');
 export const resourceQuotasCard = $('[data-test-id="resource-quotas-card"]');
+export const insightsButton = element(by.xpath('.//button[normalize-space(.)="Insights"]'));
+export const insightsPopover = element(by.xpath('.//div[@role="dialog" and .//h6[normalize-space(.)="Insights status"]]'));
+export const insightsCloseButton = insightsPopover.element(by.xpath('.//button[@aria-label="Close"]'));
```

Comment:
We should add data-test-id attributes where possible for the tests. Avoid complicated xpath selectors.

Reply:
@spadgett  There is no way to do that without changing the application code. By the way, there is a specification for locators in UI testing https://ouia.readthedocs.io/en/latest/. I added support of this in various Patternfly components https://github.com/patternfly/patternfly-react/pulls?q=is%3Apr+is%3Aclosed+ouia